week-1-ts-ecosystem/w01e03-object-oriented-structure/models/Basket.ts

```diff
@@ -0,0 +1,119 @@
+import { Product, ProductType, UpdateProductDTO } from "./Product";
+
+type NonEmptyProductArray = [Product, ...Product[]];
+
+export class Basket {
+  name: string = "";
+  #products: Product[] = [];
+  type?: ProductType;
+  public constructor(name: string);
+  public constructor(name: string, products: NonEmptyProductArray);
+  public constructor(name: string, products: undefined, type: ProductType);
```

Comment:
Ten zapis przeładowania konstruktorów - nie jest potrzebny. 
W mojej opinii - nie poprawia to czytelności.
Takie zapisy są przydatne w plikach `.d.ts` i też nie zawsze, tutaj szczegóły:

https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#use-optional-parameters